tests/kafkatest/services/console_consumer.py

```diff
@@ -256,9 +256,10 @@ def _worker(self, idx, node):
 
         consumer_output = node.account.ssh_capture(cmd, allow_fail=False)
 
-        self.init_jmx_attributes()
-        self.logger.debug("collecting following jmx objects: %s", self.jmx_object_names)
-        self.start_jmx_tool(idx, node)
+        with self.lock:
+            self._init_jmx_attributes()
```

Comment:
`start_jmx_tool` is completely generic for anything this is mixed into. `_init_jmx_metrics` is specific to `ConsoleConsumer`. That's actually why I renamed it -- it clarifies that this method is internal to this specific class.

I think we might be able to move `_init_jmx_metrics` into the constructor if we are sure that  all uses guarantee that the `JmxMixin` constructor is called with the appropriate parameters and that the choice of metrics is not modified dynamically after the service is created.